Vic2ToHoI4/Source/OutHoi4/OutHoi4World.cpp

```diff
@@ -0,0 +1,549 @@
+	 const std::string& outputName);
+void outputColorsFile(const std::map<std::string, std::shared_ptr<Country>>& countries, const std::string& outputName);
+void outputNames(const namesMapper& theNames,
+	 const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const std::string& outputName);
+void outputUnitNames(const std::map<std::string, std::shared_ptr<Country>>& countries);
+void outputMap(const States& states,
+	 const std::map<int, StrategicRegion*>& strategicRegions,
+	 const std::string& outputName);
+void outputGenericFocusTree(const HoI4FocusTree& genericFocusTree, const std::string& outputName);
+void outputCountries(const std::set<Advisor>& activeIdeologicalAdvisors,
+	 const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const allMilitaryMappings& theMilitaryMappings,
+	 const std::string& outputName);
+void outputRelations(const std::string& outputName);
+void outputLeaderTraits(const std::map<std::string, std::vector<std::string>>& ideologicalLeaderTraits,
+	 const std::set<std::string>& majorIdeologies,
+	 const std::string& outputName);
+void outputBookmarks(const std::vector<std::shared_ptr<Country>>& greatPowers,
+	 const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const std::string& outputName);
+
+} // namespace HoI4
+
+
+void HoI4::reportIndustryLevels(const World& world)
+{
+	auto militaryFactories = 0;
+	auto civilianFactories = 0;
+	auto dockyards = 0;
+	for (const auto& state: world.getStates())
+	{
+		militaryFactories += state.second.getMilFactories();
+		civilianFactories += state.second.getCivFactories();
+		dockyards += state.second.getDockyards();
+	}
+
+	LOG(LogLevel::Debug) << "Total factories: " << militaryFactories + civilianFactories + dockyards;
+	LOG(LogLevel::Debug) << "\t" << militaryFactories << " military factories";
+	LOG(LogLevel::Debug) << "\t" << civilianFactories << " civilian factories";
+	LOG(LogLevel::Debug) << "\t" << dockyards << " dockyards";
+
+	if (theConfiguration.getDebug())
+	{
+		reportCountryIndustry(world);
+		reportDefaultIndustry(world);
+	}
+}
+
+
+void HoI4::reportCountryIndustry(const World& world)
+{
+	std::ofstream report("convertedIndustry.csv");
+	report << "tag,military factories,civilian factories,dockyards,total factories\n";
+	if (report.is_open())
+	{
+		for (const auto& country: world.getCountries())
+		{
+			reportIndustry(report, *country.second);
+		}
+	}
+}
+
+
+void HoI4::reportDefaultIndustry(const World& world)
+{
+	std::map<std::string, std::array<int, 3>> countryIndustry;
+
+	for (const auto& state: world.getTheStates().getDefaultStates())
+	{
+		auto stateData = getDefaultStateIndustry(state.second);
+
+		auto country = countryIndustry.find(stateData.first);
+		if (country == countryIndustry.end())
+		{
+			countryIndustry.insert(make_pair(stateData.first, stateData.second));
+		}
+		else
+		{
+			country->second[0] += stateData.second[0];
```

Comment:
I assume operating on a country copy is fine for this case?

Reply:
find() is giving an iterator in this case, so yeah.

Also, renamed for clarity.